src/platform/crypt_bcrypt.c

```diff
@@ -764,3 +787,87 @@ CxPlatHashCompute(
 
     return NtStatusToQuicStatus(Status);
 }
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+CxPlatKbKdfDerive(
+    _In_reads_(SecretLength) const uint8_t* Secret,
+    _In_ uint32_t SecretLength,
+    _In_z_ const char* Label,
+    _In_reads_opt_(ContextLength) const uint8_t* Context,
+    _In_ uint32_t ContextLength,
+    _In_ uint32_t OutputLength,
+    _Out_writes_(OutputLength) uint8_t* Output
+    )
+{
+    BCRYPT_KEY_HANDLE KeyHandle = NULL;
+    BCryptBuffer ParameterList[3];
+    BCryptBufferDesc Parameters;
+    ULONG LabelLength = 0;
+    ULONG DerivedKeyLength;
+
+    LabelLength = (ULONG)strnlen_s(Label, 255);
+
+    NTSTATUS Status =
+        BCryptGenerateSymmetricKey(
+            CXPLAT_SP800108_CTR_HMAC_ALG_HANDLE,
```

Comment:
Question: Is this another name for KB KDF?

Reply:
Yes, this usage of EVP_KDF-KB is the same SP800-108 Counter HMAC as BCrypt. The way to switch to a different KB KDF algorithm, is to pass different parameters to EVP_KDF-KB.

Reply:
FYI https://docs.openssl.org/3.0/man7/EVP_KDF-KB/